dashboard/test/integration/pd/workshop_attendance_test.rb

```diff
@@ -1,26 +1,69 @@
 require 'test_helper'
 include Warden::Test::Helpers
 
-# More workshop attendance tests are currently in ../controllers/api/v1/pd/workshop_attendance_controller_test.rb
+# More workshop attendance tests are currently in
+# ../../controllers/api/v1/pd/workshop_attendance_controller_test.rb
 class Pd::WorkshopAttendanceTest < ActionDispatch::IntegrationTest
+  setup do
+    @organizer = create :workshop_organizer
+    @workshop = create :pd_workshop, organizer: @organizer, num_sessions: 1
+    @workshop.start!
+
+    @teacher = create :pd_workshop_participant, workshop: @workshop, enrolled: true, in_section: true
+  end
+
   # In this context and in the rails application, but not controller unit tests, empty array params are converted to nil.
   # Apparently this behavior will be reversed in Rails 5 (https://github.com/rails/rails/pull/16924),
   # and this code will correctly work either way.
   test 'handle no attendance' do
-    organizer = create(:workshop_organizer)
-    login_as organizer
-
-    workshop = create :pd_workshop, organizer: organizer
-    session = create :pd_session
-    workshop.sessions << session
-
-    patch "/api/v1/pd/workshops/#{workshop.id}/attendance", pd_workshop: {
-      session_attendances: [
-        session_id: session.id,
-        attendances: []
-      ]
-    }
+    login_as @organizer
+
+    patch attendance_url, attendance_params(false)
+
+    assert_response :success
+    assert_empty Pd::Attendance.for_workshop(@workshop).for_teacher(@teacher)
+  end
+
+  test 'update is idempotent' do
+    @workshop.start!
```

Comment:
The workshop was already started via line 10. If this is part of the idempotency test, then the test description should include start being idempotent.